addons/stock/models/stock_move.py

```diff
@@ -1404,14 +1404,12 @@ def _assign_picking_values(self, picking):
         if any(picking.partner_id != m.partner_id for m in self):
             vals['partner_id'] = False
         if any(picking.origin != m.origin for m in self):
-            current_origins = set(picking.origin.split(',') + [False]) if picking.origin else {False}
             vals['origin'] = picking.origin
-            for move in self:
-                if move.origin not in current_origins:
-                    if not vals['origin']:
-                        vals['origin'] = move.origin
-                    else:
-                        vals['origin'] += f',{move.origin}'
+            current_origins = picking.origin.split(',') if picking.origin else []
+            new_origins = [move.origin for move in self if move.origin]
+            all_origins = OrderedSet(filter(None, current_origins + new_origins))
```

Comment:
I might be mistaken but I do not think the filter is useful here:
```suggestion
            all_origins = OrderedSet(current_origins + new_origins)
```